### DISCUSSION TITLE
NS on Google Cloud VM: Set up automatic security updates?

### DISCUSSION THREAD

**Discussion by bjornoleh** [NightscoutFoundation/xDrip#2461] [Ideas]
Hi,

should we set up automatic security updates on our Compute Engine VMs using "unattended-upgrades" or similar?

https://help.ubuntu.com/community/AutomaticSecurityUpdates

I would imagine some (most?) people will not be SSH-ing into their VMs very often to update/upgrade. Perhaps some automation here is the safest strategy? This is obviously at the risk of breaking something unattended, but perhaps this is still a good idea if implemented conservatively? I am quite new to Linux, have just tried to read up a bit. 

Thanks for any pointers about what is a reasonable strategy for this. And perhaps it can be included in the setup script eventually?

**Comment by Navid200** [comment]
When something required on update, it was announced in forums and people had to log into Heroku to update.

Now, when something comes up that requires an update, we will announce in these forums and people can log into their virtual machine, and follow our instructions to update.

I don't see why anything has to change in that respect.

**Reply by bjornoleh** [reply]
Sure, that's an option of course. Experience does however tell that a considerable number of users only visit the forums once their system is no longer working. We'll see on November 28 when the Heroku apps are taken down. I forecast a surge in activity on the forums then... Some automated updates running silently would perhaps take good care of this group.

I don't know if the Linux updates are critical enough for it to matter much for the reliability and safety of Nightscout installations. Perhaps it hardly matters at all?

Nightscout updates would also be possible to automate, especially if using Docker. But there could certainly be valid reasons not to do that, such as the possibility of a broken release.
